src/util.jl

```diff
@@ -162,11 +163,18 @@ Convert an amplitude ratio to dB (decibel), or ``20
 amp2db(a::Real) = 20*log10(a)
 
 """
-    rms(s)
+    rms(s; dims)
 
-Return the root mean square of signal `s`.
+Return the root mean square (rms) of signal `s`. Optional keyword parameter
+`dims` can be used to specify the dimensions along which to compue the rms.
 """
-rms(s::AbstractArray{T}) where {T<:Number} = sqrt(sum(abs2, s)/length(s))
+function rms(s::AbstractArray{T}; dims=:) where {T<:Number}
+    if dims === (:)
+        return sqrt(mean(abs2, s))
+    else
+        return sqrt.(mean(abs2, s; dims=dims))
```

Comment:
This would also work if `dims === (:)`, but that assumes that the result `Number` of `mean` broadcasts as a singleton, which I didn't feel 100% comfortable relying on, therefore the branch to avoid the broadcast when `dims === (:)`.

Reply:
Well, also `mean(f, X; dims)` doesn't exist on Julia 1.0, so this needs a special case anyway.